### PR TITLE
[SYCL] Separate OpenCL specific constructs from lit tests.

### PR DESCRIPTION
**Pull request by rbegam**
Adding 'opencl' requirement disabled several tests some of which
have non-opencl constructs as well. This patch separates those tests
to enable testing of SYCL core functionalities.

Signed-off-by: rbegam <rehana.begam@intel.com>

### CODE REVIEW

File: sycl/test/inorder_queue/in_order_buffs.cpp [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// REQUIRES: opencl
-
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out -L %opencl_libs_dir -lOpenCL
```
Comment (vladimirlaz) [inline sycl/test/inorder_queue/in_order_buffs.cpp:1]:
```suggestion
// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out

File: sycl/test/basic_tests/event.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// REQUIRES: opencl
+// REQUIRES: opencl || level0
```
Comment (vladimirlaz) [inline sycl/test/basic_tests/event.cpp:1]: I wonder how clCreateUserEvent call below will be processed if only Level0 RT is available

File: sycl/test/inorder_queue/in_order_dmemll.cpp [hunk 3]
```diff
@@ -1,5 +1,3 @@
-// REQUIRES: opencl
-
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t1.out -L %opencl_libs_dir -lOpenCL
```
[the remark this answers is not in the document]
Comment (rbegam) [reply sycl/test/inorder_queue/in_order_dmemll.cpp:1]: correct. will fix it.

File: sycl/test/basic_tests/event.cpp (hunk 2 above)
Comment (rbegam) [reply sycl/test/basic_tests/event.cpp:1]: clCreateUserEvent is not called for host and the test sets SYCL_DEVICE_TYPE=HOST. Do you suggest separating this test too?

**Comment by rbegam** [comment]
ping @smaslov-intel @vladimirlaz

File: sycl/test/basic_tests/event.cpp (hunk 2 above)
Comment (vladimirlaz) [reply sycl/test/basic_tests/event.cpp:1]: I am not quite understand the logic of the test:
 1. if we force HOST device lines 26-31 are dead code.
 2. It looks like the whole test check host events. What is the reason for having OpenCL or Level0 BE?

**Comment by vladimirlaz** [comment]
> ping @smaslov-intel @vladimirlaz

see my comment above

File: sycl/test/basic_tests/event.cpp [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
```
Comment (bader) [inline sycl/test/basic_tests/event.cpp:1]: https://github.com/intel/llvm/commit/215f591ade97aed616c4094a467e5dcbebc45d6b changed this feature name.

level0 -> level_zero

Please, fix ASAP.

Comment (rbegam) [reply sycl/test/basic_tests/event.cpp:1]: up for review https://github.com/intel/llvm/pull/2235

**Pull request merged**